instrumentation/spring-webmvc/src/main/java/brave/spring/webmvc/TracingHandlerInterceptor.java

```diff
@@ -68,6 +82,16 @@ public void afterCompletion(HttpServletRequest request, HttpServletResponse resp
     Span span = tracer.currentSpan();
     if (span == null) return;
     ((SpanInScope) request.getAttribute(SpanInScope.class.getName())).close();
-    handler.handleSend(response, ex, span);
+    Object template = request.getAttribute(BEST_MATCHING_PATTERN_ATTRIBUTE);
+    if (template == null) { // skip thread-local overhead if there's no attribute
```

Comment:
ps I know this stuff is mostly copy/paste between here and vert.x (except here the attribute is an object not a string). I'm letting that pressure build before generalizing code